My Home: Promote the menu editing tutorial as the primary action in the "Edit the site menu" task

p-jackson opened the pull request:
The site setup checklist on My Home includes a task to "edit your menu". Completing this task means taking users to the customiser, which can be confusing.

This change is discussed in greater detail in #47334, but in short this PR encourages users to view the tutorial for menus over actually making the menu change. It also replaces the explicit "skip" action with an implicit skippability. Now if the only task left incomplete is the menu task, then it will be counted as skipped. This depends on the change in D61706-code, and is a bit of a hack that tries to make up for the fact that the "edit your menu" task has lost its skip button.

#### Changes proposed in this Pull Request

* Allow inline support links to be custom styled
* Move the menu tutorial link from the task description to the primary action
* Render the "Add a menu" button as a secondary action
* Hide the skip button from the task
* Make the task not count towards setup completion

<img width="951" alt="Screenshot 2021-05-19 at 3 30 28 PM" src="https://user-images.githubusercontent.com/1500769/119066117-2ca45e00-ba33-11eb-9b8a-354e2d16e00e.png">

The secondary "Add a menu" button still fires the `calypso_checklist_task_start` track event.
The "View tutorial" button still fires the `calypso_customer_home_menus_support_page_view` track event and still opens in a modal.

#### Testing instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Test using calypso live and a site that has not completed setup yet
* Test buttons still work as expected
* Apply D61706-code
* Create a new unlaunched site
  * Complete every task except editing the menu and launching
  * Launch the site
  * When landing on the My Home page notice the site setup is complete
  * Reload page with `?disable-nav-unification` and see that the site setup progress bar isn't visible
* Create another new unlaunched site
  * Launch the site before completing any other tasks
  * After landing on My Home page get celebration that site was launched
  * Dismiss that celebration notice by saying you want to continue setup
  * See that the checklist is still there

<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Related to #47334

p-jackson committed:
<InlineSupportLink> supports overriding `className` prop

p-jackson committed:
Style "Add a menu" task as a secondary link

p-jackson committed:
Make the menu tutorial link the primary action in setup checklist

autumnfjeld reviewed (dismissed):
✅ Working as expected.

BEFORE:
![image](https://user-images.githubusercontent.com/5665959/118906140-84be6000-b960-11eb-903e-033fde9b831b.png)

AFTER:
![image](https://user-images.githubusercontent.com/5665959/118906042-47f26900-b960-11eb-8fa5-041b455ed70a.png)

✅ As long as discussion is settled over in https://github.com/Automattic/wp-calypso/issues/47334#issuecomment-843725187

p-jackson committed:
Remove skip option from menu editing task

p-jackson committed:
The edit menu task doesn't count towards completion of site setup

Now that the task isn't skippable, it can block setup completion (all
steps need to be completed before the checklist will disappear, even
post launch).

autumnfjeld commented:
@simison & @davemart-in  We'd like to proceed with shipping this PR,  as I explained in the https://github.com/Automattic/wp-calypso/issues/47334#issuecomment-847496302.  Just flagging you in case you see any issues we might not have considered.

p-jackson commented:
Copy/pasting video from https://github.com/Automattic/wp-calypso/issues/47334#issuecomment-845585733 of the new behaviour for users who have done all checklist tasks other than "edit menu"

https://user-images.githubusercontent.com/1500769/119437827-0b18de80-bd73-11eb-9e0c-d8bf39fcf7a7.mov

davemart-in commented:
> @simison & @davemart-in We'd like to proceed with shipping this PR

Looking good! 👍
